File: lib/src/main/java/com/sd/lib/cache/handler/impl/StringHandler.java
package com.sd.lib.cache.handler.impl;

import com.sd.lib.cache.CacheInfo;
import com.sd.lib.cache.handler.BaseCacheHandler;

/**
 * String处理类
 */
public class StringHandler extends BaseCacheHandler<String> {
    public StringHandler(CacheInfo info) {
        super(info);
    }

    @Override
    protected byte[] valueToByte(String value) {
        return value.getBytes();
    }

    @Override
    protected String byteToValue(byte[] bytes, Class<?> clazz) {
        return new String(bytes);
    }

    @Override
    protected String getKeyPrefix() {
        return "string_";
    }
}

File: lib/src/main/java/com/sd/lib/cache/handler/impl/StringHandler.kt
package com.sd.lib.cache.handler.impl

import com.sd.lib.cache.CacheInfo
import com.sd.lib.cache.handler.BaseCacheHandler

/**
 * String处理类
 */
internal class StringHandler(info: CacheInfo) : BaseCacheHandler<String>(info, "string") {
    override fun encodeToByteImpl(value: String): ByteArray {
        return value.toByteArray()
    }

    override fun decodeFromByteImpl(bytes: ByteArray, clazz: Class<*>?): String {
        return String(bytes)
    }
}